#!/usr/bin/env python

# This configuration file is for the rqt panel that is used in
# dynamic reconfigure.

# author: subramak@andrew.cmu.edu
# Date  : 4 Sep 2020

# Changelog:
# 	subbu - 09/04 - Initial commit


PACKAGE = "autovalet_husky"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Actual TF params:
gen.add("x_pos", double_t, 0, "X_Position",    0, -1,   1)		# defaults to 0 and ranges from -1 to 1
gen.add("y_pos", double_t, 0, "Y_Position",    0, -1,   1) 	    # default 0 and ranges from -1 to 1
gen.add("z_pos", double_t, 0, "Z_Position",    0, -1,   1)		# default 0 and ranges from -1 to 1


gen.add("yaw", double_t, 0, "Yaw",    0, -3.14,   3.14)			# default 0 and ranges from -2*pi to +2*pi
gen.add("pitch", double_t, 0, "Pitch",    0, -3.14,   3.14)		# default 0 and ranges from -2*pi to +2*pi
gen.add("roll", double_t, 0, "Roll",    0, -3.14,   3.14)		# default 0 and ranges from -2*pi to +2*pi

exit(gen.generate(PACKAGE, "vlp_cam_calib", "lidar_camera_tf"))
